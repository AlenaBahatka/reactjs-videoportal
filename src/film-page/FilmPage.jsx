import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import FilmList from '../common-components/film-list/FilmList';
import FilmToolbar from './toolbar/FilmToolbar';
import FilmHeader from './header/FilmHeader';
import Footer from '../common-components/footer/Footer';
import NetflixLabel from '../common-components/labels/NetflixLabel';

import * as actions from '../+state/actions/actions';

export class FilmPage extends PureComponent {
	componentDidMount() {
		let filmId = 411741; // TODO: should be taken from url later
		this.props.getFilmWithSimilar(`https://reactjs-cdp.herokuapp.com/movies/${filmId}`);
	}

	render() {
		let { film, similarFilms } = this.props;
		return (
			<div className="panel panel-default">
				<div className="panel-heading">
					<div>
						<NetflixLabel />
						<button>Search</button>
					</div>
					<FilmHeader filmInfo={film} />
				</div>
				<div className="panel-body">
					<FilmToolbar genre={film.genre} />
					<FilmList films={similarFilms} />
				</div>
				<Footer />
			</div>
		);
	}
}

FilmPage.propTypes = {
	film: PropTypes.object,
	similarFilms: PropTypes.array
};

const mapStateToProps = (state) => {
	return {
		film: state.selectedFilm,
		similarFilms: state.similarFilms
	};
};

export default connect(mapStateToProps, actions)(FilmPage);
