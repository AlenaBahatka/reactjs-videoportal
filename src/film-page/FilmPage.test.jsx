import React from 'react';
import { shallow, mount } from 'enzyme';
import FilmPage from './FilmPage';

let film = {},
	similarFilms = [];

describe('FilmPage', () => {
	it('should correctly render component', () => {
		let filmPageComponent = shallow(<FilmPage film={film} similarFilms={similarFilms} />);

		expect(filmPageComponent).toMatchSnapshot();
	});
});
