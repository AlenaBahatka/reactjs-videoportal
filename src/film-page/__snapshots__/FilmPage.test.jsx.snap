// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPage should correctly render component 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <div>
      <NetflixLabel />
      <button>
        Search
      </button>
    </div>
    <FilmHeader
      filmInfo={Object {}}
    />
  </div>
  <div
    className="panel-body"
  >
    <FilmToolbar />
    <FilmList
      films={Array []}
    />
  </div>
  <Footer />
</div>
`;
